WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt

```diff
@@ -38,12 +44,20 @@ class ScanViewModel @Inject constructor(
         }
         isStarted = true
         this.site = site
-        init()
+        fetchScanState()
     }
 
-    private fun init() {
-        scanStatusService.scanState.observeForever(scanStateObserver)
-        scanStatusService.start(site)
+    private fun fetchScanState() {
+        launch {
+            fetchScanStateUseCase.fetchScanState(site = site)
+                .flowOn(bgDispatcher)
```

Comment:
🔍 Nitpick: We might want to consider setting the dispatcher in the usecase so no-one can invoke the usecase on the ui thread. Wdyt?

Reply:
Nice suggestion 👍. 
In 79b6480